### DISCUSSION TITLE
Diminishing disk space on Ubuntu images

### DISCUSSION THREAD

**Discussion by saudet** [community/community#26493] [Actions]
<p>Our Linux builds have started to fail because they are running out of disk space. While investigating the cause of the failures, I found that the amount of disk space on the Ubuntu images are declining on a weekly basis. Here is what gets displayed in our log for <code>df -h</code> within a typical Docker container that we’re using for our builds that takes about 4 GB of disk space once everything installed:</p>
<p><a href="https://github.com/tensorflow/java/runs/515298915" rel="nofollow noopener">https://github.com/tensorflow/java/runs/515298915</a> (Mar 18, 2020)</p>
<pre><code>overlay84G72G12G86% /
</code></pre>
<p><a href="https://github.com/tensorflow/java/runs/559103432" rel="nofollow noopener">https://github.com/tensorflow/java/runs/559103432</a> (Apr 4, 2020) </p>
<pre><code>overlay84G75G8.8G90% /
</code></pre>
<p><a href="https://github.com/tensorflow/java/runs/581350731" rel="nofollow noopener">https://github.com/tensorflow/java/runs/581350731</a> (Apr 13, 2020)</p>
<pre><code>overlay84G76G7.8G91% /
</code></pre>
<p>So it looks like we’re losing about 1 GB of space every week, and in about 3 months, we’ll be reaching 0 GB, probably preventing the images from being able to boot at all.</p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/saudet">@saudet</a> ,</p>
<p>I checked your workflow yml file, you are using container nvidia/cuda:10.1-cudnn7-devel-centos7 to run your job. </p>
<p>Based on my test , the decreased disk space is used by this image. Please see my job yml and workflow logs :</p>
<pre><code>job1:
    runs-on: ubuntu-latest
    steps:
      - name: df
        run: |
          df -h
          docker system df
          docker run nvidia/cuda:10.1-cudnn7-devel-centos7
          df -h
          docker system df
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a86692ae783527ca4a5197637a281a0f3c1a1306.png" data-download-href="/uploads/short-url/o1JYy7W7ReGR17m33Se5UpDWvzg.png?dl=1" title="before.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181098025-0d9068bf-71f3-42e0-a36b-c72507ed61e9.png" alt="before.png" data-base62-sha1="o1JYy7W7ReGR17m33Se5UpDWvzg" width="638" height="499" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a86692ae783527ca4a5197637a281a0f3c1a1306_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">before.png</span><span class="informations">733×574 13.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a314fafd5620b273e5e4a8a180337a27425acc22.png" data-download-href="/uploads/short-url/ngGMeNsmCLiR0Wi4rBJhX8rrA66.png?dl=1" title="after.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181098027-baa393f9-f5a1-4f19-acb8-17a3d78bcb84.png" alt="after.png" data-base62-sha1="ngGMeNsmCLiR0Wi4rBJhX8rrA66" width="690" height="492" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a314fafd5620b273e5e4a8a180337a27425acc22_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">after.png</span><span class="informations">757×540 14.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The size of image is 3.5GB . </p>

**Comment by saudet** [comment]
<p>Yes, I know and I did mention that the container takes about 4 GB of space. The point is, 13 GB of space is less than the 14 GB that is supposed to be guaranteed:</p>
<p>Each virtual machine has the same hardware resources available.</p>
<ul>
<li>2-core CPU</li>
<li>7 GB of RAM memory</li>
<li>14 GB of SSD disk space</li>
</ul>
<p><a href="https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners" rel="nofollow noopener">https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners</a></p>
<p>If you only guarantee 13 GB, then the documentation should be updated. Do you agree?</p>

**Comment by saudet** [comment]
<p>As of at least yesterday, we’re now down to about 9 GB:</p>
<p><a href="https://github.com/tensorflow/java/runs/599744540" rel="nofollow noopener">https://github.com/tensorflow/java/runs/599744540</a> (Apr 20, 2020)</p>
<pre><code>overlay 84G 80G 3.7G 96% /
</code></pre>
<p>I doubt very much we’re supposed to have so little disk space to work with.</p>

**Comment by subotic** [comment]
<p>I had seen the same thing on our builds that started to fail. The following values are after <code>docker system prune</code>:</p>
<p><span>/dev/sda1    84G    74G    9.9G    89%    /</span></p>
<p>And today:</p>
<p><span>/dev/sda1    84G    70G    15G    84%    /</span></p>
<p>So they fixed whatever they temporarely broke (I hope). It would have been nice to acknowledge the problem, so that we now that a fix was underway.</p>

**Comment by Yanjingzhu** [comment]
<p>You could see the logs in my above screenshot , the /dev/sdb1 is 14 G, used 41M, then the Avail show 13G . It is not an accurate data. </p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Sorry for this bad experience. There is an opening issue in virtual-environments repo ,  engineering team are working on a fix fot it. <a href="https://github.com/actions/virtual-environments/issues/709#issuecomment-616767507" rel="nofollow noopener">https://github.com/actions/virtual-environments/issues/709#issuecomment-616767507 </a></p>
<p>Please wait for sometime. Kindly let me know your current status. </p>
